### ISSUE TITLE
data in "before create" hooks is empty

### ISSUE THREAD

**Issue by invakid404** [pocketbase/pocketbase#80]
While playing around with using PocketBase as a framework, I noticed that the `User` in the `OnUserBeforeCreateRequest` hook is empty. Let's take the following hook as an example:

```go
	app.OnUserBeforeCreateRequest().Add(func(event *core.UserCreateEvent) error {
		encoder := json.NewEncoder(os.Stdout)
		encoder.SetIndent("", "  ")

		return encoder.Encode(event.User)
	})
```

When I create a user, I get the following output:

![image](https://user-images.githubusercontent.com/38114607/178345469-fdef62d0-b7ba-409c-a119-463ba171280b.png)

All fields appear to be empty. By stepping through the "user create" endpoint with a debugger, I observed this:
1) The request body is parsed correctly in the `form`, but
2) Those fields aren't set in the user object yet

![image](https://user-images.githubusercontent.com/38114607/178345730-64db0244-9aef-47e6-a1e8-d977bfdba9df.png)

Reading through the code, I confirmed that the user struct's fields don't get touched until `Submit()` is called on the form: [here](https://github.com/pocketbase/pocketbase/blob/3d07f02/forms/user_upsert.go#L106).

And since `Submit()` isn't called until the `OnUserBeforeCreateRequest` hooks have finished executing, the hooks don't get access to the user that's about to be created.

This definitely doesn't sound like expected behavior (otherwise that hook would be very useless indeed). Without really knowing how this project is structured, it seems like the field setting portion of `Submit()` should happen earlier somewhere.

This appears to be the case with other "before create" hooks. I also tried the "OnRecordBeforeCreateRequest" hook, but the record in there has no data whatsoever either, besides the "@collectionId" and "@collectionName" fields. Calling `.Data()` on it doesn't help much either, a blank map is returned.

**Comment by ganigeorgiev** [comment]
Thank you for the detailed report. This is indeed is a bug that I've introduced a week before the release (initially the events were operating on the form instance which is populated before triggering the event).

It'll be fixed in the next v0.3.0 version (together with the nullable field controls).

**Comment by ganigeorgiev** [comment]
I've decided to release the fix for this in [v0.2.4](https://github.com/pocketbase/pocketbase/releases) because it could be a blocker if someone is using PocketBase as a framework.

The form request unit tests were update to reflect the change, but we'll have to think how to improve the integration tests for the api, because currently we are verifying only whether the hook was triggered or not, but there are is no easy way to test the hook(s) data.

@invakid404 Please test it and if you stumble on something, feel free to let me know and I'll investigate it further.
